src/sqlfluff/core/templaters/python.py

```diff
@@ -739,17 +739,17 @@ def _split_uniques_coalesce_rest(
                 # recursion.
                 if len(int_file_slice.slice_buffer) > bookmark_idx:
                     # Recurse to deal with any loops separately
-                    sub_section = int_file_slice.slice_buffer[
-                        bookmark_idx : len(int_file_slice.slice_buffer)
-                    ]
                     yield from cls._split_uniques_coalesce_rest(
                         [
                             IntermediateFileSlice(
                                 "compound",
                                 # Slicing is easy here, we have no choice
                                 slice(starts[0], int_file_slice.source_slice.stop),
                                 slice(starts[1], int_file_slice.templated_slice.stop),
-                                sub_section,
+                                # Calculate the subsection to deal with.
+                                int_file_slice.slice_buffer[
+                                    bookmark_idx : len(int_file_slice.slice_buffer)
+                                ],
```

Comment:
I've totally removed the `sub_section` variable here, it having the same name as the other patch is confusing.